python/tank/hook.py

```diff
@@ -246,10 +247,22 @@ def get_publish_paths(self, sg_publish_data_list):
                                 "not contain a valid path definition" % sg_data)
 
             local_path = path_field.get("local_path")
-            if local_path is None:
+            url = path_field.get("url")
+            if local_path:
```

Comment:
I think toolkit should try to do the best it can with the data that is delivered from the server. In the case both local path and url are returned, local path should take precedence. I'll add a check and a warning!